### ISSUE TITLE
No Cognito Federated Identity pool provided when using withAuthenticator

### ISSUE THREAD

**Issue by magjack** [aws-amplify/amplify-js#5528]
**Describe the bug**
When using withAuthenticator(app, {federated: federateConfig}) my user is not being authenticated after a successful login with federated buttons and returns to the signIn state of the app. 

I get a error code in the console. : 
![image](https://user-images.githubusercontent.com/40371034/80157984-520c3180-85bf-11ea-921f-78f7fa5d4dcf.png)

**To Reproduce**
Create an app with federated auth. 
add withAuthenticator(app, {federated: federateConfig}) in index.js
Click log in with google
Successfully login in with a gmail account. 

result is landing back at the login screen. 

**Expected behavior**
App should authenticate the user and create a user in cognito pull

**Code Snippet**
The only place this error is produced is in @aws-amplify/core/src/Credentials.js
![image](https://user-images.githubusercontent.com/40371034/80158590-c5fb0980-85c0-11ea-8cdf-22d2d89acefd.png)

And the only cause is a missing identityPoolId string which I can see is set in my amplify-meta.json
![image](https://user-images.githubusercontent.com/40371034/80158708-13777680-85c1-11ea-851d-5501a6fdb7a5.png)

This is all managed by amplify so I think there is a disconnect between these values.
I have confirmed federation is configured 
![image](https://user-images.githubusercontent.com/40371034/80159309-8e8d5c80-85c2-11ea-95f3-5bc84e3e66ff.png)

**What is Configured?**
If applicable, please provide what is configured for Amplify CLI:
* Which steps did you follow via Amplify CLI when configuring your resources.
amplify add auth
* Which resources do you have configured?
auth, graphql, functions

  
```
const awsmobile = {
    "aws_project_region": "eu-west-1",
    "aws_cognito_region": "eu-west-1",
    "aws_user_pools_id": "eu-west-1_6",
    "aws_user_pools_web_client_id": "d1",
    "oauth": {
        "domain": "lll",
        "scope": [
            "phone",
            "email",
            "openid",
            "profile",
            "aws.cognito.signin.user.admin"
        ],
        "redirectSignIn": "lllll",
        "redirectSignOut": "lllll",
        "responseType": "code"
    },
    "federationTarget": "COGNITO_USER_POOLS",
    "aws_appsync_graphqlEndpoint": "llllll",
    "aws_appsync_region": "eu-west-1",
    "aws_appsync_authenticationType": "AMAZON_COGNITO_USER_POOLS",
    "aws_cloud_logic_custom": [
        {
            "name": "lll",
            "endpoint": "lll",
            "region": "eu-west-1"
        }
    ]
};
```

```
{
  "name": "my-app",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@aws-amplify/ui-react": "^0.2.3",
    "@material-ui/core": "^4.2.1",
    "@material-ui/icons": "^4.2.1",
    "aws-amplify": "^3.0.8",
    "bootstrap": "^4.3.1",
    "lodash": "^4.17.15",
    "react": "^16.8.6",
    "react-bootstrap": "^1.0.0-beta.10",
    "react-dom": "^16.8.6",
    "react-router-dom": "^5.0.1",
    "react-scripts": "^3.4.0"
  }
```

Amplify CLI -v 
4.18.1

**Smartphone (please complete the following information):**

- Device: Acer laptop
- OS: Windows ten 
- Browser: Chrome, FireFox 
- Versio: Version 81.0.4044.113, 75

**Additional context**

**_You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app._**

**Comment by elorzafe** [comment]
@magjack If you are using Social Federation with Cognito User Pools you should use `withOAuth` HOC instead of `withAuthenticator`.

More info you can find [here](https://docs.amplify.aws/lib/auth/social/q/platform/js#facebook-instructions)

**Comment by magjack** [comment]
I managed to do a few work arounds to get the federated buttons to grant a user access to the app however they are unable to use it any further because a user is not created as part of the sign in. 
Work arounds where: 

I needed to add this config to my app 

`Auth.configure({
  identityPoolId: 'eu-west-1:dfggrr-fffde-eert-yuu-eerrt45r'
})`
This was already under the aws-exports.js as aws_cognito_identity_pool_id

I also had to manually update the federated Identity pool created on the cognito console with the app IDs as I was getting another error 'token is not from a supported provider of this identity pool'

I see this is a common [issue ](https://github.com/aws-amplify/amplify-js/issues/3875)

I will investigate using the withOAuth HOC however this will require a bit of redesign in the app. 

I am unclear on why the withAuthenticator function has the ability to take a federated config and display the sign in buttons but not then be able to sign in and create a user. Will there be a release soon that will allows this ? I also noticed that the withAuthenticator was moved to a new package @aws-amplify/ui-react however withOAuth is now not part of that, will it be soon. Also with the withOAuth method do I not lose the built in UI components that come with withAuthenticator. Is there a hybrid option of withOauth and withAuthenticator?
